ui/raidboss/data/04-sb/raid/o3n.js

```diff
@@ -0,0 +1,474 @@
+[{
+  zoneRegex: {
+    en: /^Deltascape \(V3\.0\)$/,
+  },
+  timelineFile: 'o3n.txt',
+  timelineTriggers: [
+    {
+      id: 'O3N Frost Breath',
+      regex: /Frost Breath/,
+      beforeSeconds: 4,
+      condition: function(data) {
+        return data.role == 'tank'&& !data.activeTank == data.me;
+      },
+      // This will at least warn the Dragon tank that they need to be careful.
+      response: Responses.tankCleave('alert'),
+    },
+  ],
+  triggers: [
+    {
+      id: 'O3N Phase Initialization',
+      regex: Regexes.ability({ id: '367', source: 'Halicarnassus', capture: false }),
+      regexDe: Regexes.ability({ id: '367', source: 'Halikarnassos', capture: false }),
+      regexFr: Regexes.ability({ id: '367', source: 'Halicarnasse', capture: false }),
+      regexJa: Regexes.ability({ id: '367', source: 'ハリカルナッソス', capture: false }),
+      regexCn: Regexes.ability({ id: '367', source: '哈利卡纳苏斯', capture: false }),
+      regexKo: Regexes.ability({ id: '367', source: '할리카르나소스', capture: false }),
+      condition: function(data) {
+        return !data.phaseNumber;
+      },
+      run: function(data) {
+        data.phaseNumber = 1;
+      },
+    },
+    {
+      id: 'O3N Phase Tracker',
+      regex: Regexes.startsUsing({ id: '2304', source: 'Halicarnassus', capture: false }),
+      regexDe: Regexes.startsUsing({ id: '2304', source: 'Halikarnassos', capture: false }),
+      regexFr: Regexes.startsUsing({ id: '2304', source: 'Halicarnasse', capture: false }),
+      regexJa: Regexes.startsUsing({ id: '2304', source: 'ハリカルナッソス', capture: false }),
+      regexCn: Regexes.startsUsing({ id: '2304', source: '哈利卡纳苏斯', capture: false }),
+      regexKo: Regexes.startsUsing({ id: '2304', source: '할리카르나소스', capture: false }),
+      run: function(data) {
+        data.phaseNumber += 1;
+      },
+    },
+    {
+      // Normal spellblade holy with one stack point.
+      // "64" is a stack marker.  "65" is the prey marker.
+      // The debuff order in the logs is:
+      //   (1) stack marker
+      //   (2) prey marker
+      //   (3) prey marker
+      id: 'O3N Spellblade Holy Standard',
+      regex: Regexes.headMarker({ id: ['0064', '0065'] }),
+      condition: function(data, matches) {
+        // Cave phase has no stack markers.
+        if (data.phaseNumber == 2)
+          return false;
+
+        data.holyTargets = data.holyTargets || [];
+        data.holyTargets.push(matches.target);
+        return data.holyTargets.length == 3;
+      },
+      alertText: function(data) {
+        if (data.holyTargets[0] == data.me) {
+          return {
+            en: 'Stack on YOU',
+            de: 'Stack auf DIR',
+          };
+        }
+        for (let i = 1; i < 3; i++) {
+          if (data.holyTargets[i] == data.me) {
+            return {
+              en: 'Get out',
+              de: 'Raus da',
+            };
+          }
+        }
+        return {
+          en: 'Stack on ' + data.holyTargets[0],
```

Comment:
Can you fill in the other "stack on" translations from responses.js? Same with Get Out and Spread?

One other option here is to make this a `response:` instead of an `alertText` and do something like,

```
if (data.holyTargets[i] == data.me)
  return Responses.getOut()(data, matches);
```

I want to think of a better way to handle calling Responses from inside a response function, but I think this is still an option.  I don't think that there's any great way at the moment to handle stack, unless you wrote your own matches object and then called Responses.stackOn()(data, {target: data.holyTaragets[0]}), but that seems a bit invasive too.  I'm still trying to figure out the best way to handle all of this.

Reply:
That's fair. I think for this specific case I'll just enumerate the translations from `responses.js, since I don't have time over lunch to test and bugfix using `response`. I'll probably follow that up with a proper fix later. "Sometime".

Reply:
That's fair for now.  There was a similar conversation in #1269, so maybe it'd be good to just have some set of functions that you can call that just return a `{ en: 'stuff here' }` object vs a nested `{ alertText: { en: 'stuff here' } }` style object.

Reply:
(Commenting here so as to avoid cluttering the discussion on the other request, and I don't think this really warrants a separate issue from me.)

Naively, what about adding `ResponseText` to `responses.js`?

```
let Responses = {
  stackOn: (sev) => {
    let obj = {};
    obj[defaultAlertText(sev)] = (data, matches) => {
      let target = getTarget(matches);
      if (target == data.me) {
        return responseText.stackOnYou();
      }
      return responseText.stackOnTarget(data, target);
    };
    return obj;
  },
}
let responseText = {
  stackOnYou: () => {
    return {
      en: 'Stack on YOU',
      de: 'Auf DIR sammeln',
      fr: 'Package sur VOUS',
      ja: '自分にスタック',
      cn: '集合点名',
      ko: '쉐어징 대상자',
    };
  },
  stackOnTarget: (data, target) => {
    return {
      en: 'Stack on ' + data.ShortName(target),
      de: 'Auf ' + data.ShortName(target) + ' sammeln',
      fr: 'Package sur ' + data.ShortName(target),
      ja: data.ShortName(target) + 'にスタック',
      cn: '靠近 ' + data.ShortName(target) + '集合',
      ko: '쉐어징 → ' + data.ShortName(target),
    };
  },
},
```
(I'm not *terribly* happy about layering Yet More Functions over things? But it would maintain consistency with the rest of the helper constructs.)

I feel like it kind of makes sense to have the translation objects live in the same file as the response functions. If not, we could make `common-translations.js` or something and dump all the `responseText` objects in there.